src/fosslight_util/_get_downloadable_url.py

```
@@ -13,6 +13,109 @@
+        for href in source_links:
+            abs_url = _absolute_packages_debian_url(href)
+            if "deb.debian.org/debian/pool/" in abs_url:
+                return abs_url.replace("https://deb.debian.org/debian/pool/", "http://deb.debian.org/debian/pool/")
+
+        # If relative path to /debian/pool is provided, normalize to deb.debian.org.
+        for href in source_links:
+            abs_url = _absolute_packages_debian_url(href)
+            if "/debian/pool/" in abs_url:
+                suffix = abs_url.split("/debian/pool/", 1)[1]
+                return f"http://deb.debian.org/debian/pool/{suffix}"
+    except Exception as e:
```

Ruff (0.15.11)

[warning] Do not catch blind exception: `Exception`

(BLE001)